src/jsonrpc/api.rs

```diff
@@ -136,6 +136,50 @@ pub fn handle_request(control: &DaemonControl, req: Request) -> Result<Response,
                 .ok_or_else(|| Error::invalid_params("Missing 'psbt' parameter."))?;
             update_spend(control, params)?
         }
+        "listconfirmed" => match req.params {
+            Some(Params::Array(params)) => {
+                if params.len() != 3 {
+                    return Err(Error::invalid_params("command requires 3 parameters"));
+                }
+                if let (Some(start), Some(end), Some(limit)) = (
+                    serde_json::Value::as_u64(&params[0]),
+                    serde_json::Value::as_u64(&params[1]),
+                    serde_json::Value::as_u64(&params[2]),
+                ) {
+                    serde_json::json!(&control.list_confirmed_transactions(
+                        start as u32,
+                        end as u32,
+                        limit
+                    ))
+                } else {
+                    return Err(Error::invalid_params("command requires 3 parameters"));
+                }
+            }
+            None => return Err(Error::invalid_params("command requires 3 parameters")),
+            _ => return Err(Error::invalid_params("invalid parameters")),
```

Comment:
This is incorrect. See how it's done for the other commands above. Users must be able to pass parameters either as an array or as a mapping. There a `get()` helper specifically for this purpose.

While you are at it, it's probably better to make those into their own functions like it's done for all other commands that need to parse parameters.